Spigot-Server-Patches/0267-PlayerAdvancementGrantEvent.patch

```diff
@@ -0,0 +1,30 @@
+From 2745f6a39a9b9563c3c46a20e354bd1d45150732 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 19 Jan 2018 08:15:29 -0600
+Subject: [PATCH] PlayerAdvancementGrantEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/AdvancementDataPlayer.java b/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
+index 6896b709..e9eacae2 100644
+--- a/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
++++ b/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
+@@ -198,6 +198,16 @@ public class AdvancementDataPlayer {
+         if (advancementprogress.a(s)) {
+             this.d(advancement);
+             this.i.add(advancement);
++            // Paper start
++            com.destroystokyo.paper.event.player.PlayerAdvancementGrantEvent advancementGrantEvent = new com.destroystokyo.paper.event.player.PlayerAdvancementGrantEvent(this.player.getBukkitEntity(), advancement.bukkit, s);
++            this.player.world.getServer().getPluginManager().callEvent(advancementGrantEvent);
++            if (advancementGrantEvent.isCancelled()) {
```

Comment:
This line, and the line above could be simplified to `advancementGrantEvent.callEvent()`. This would remove some boilerplate code.

Reply:
Good call. I didn't know that was even a thing ^_^